### PR TITLE
PR for array_of_similar_arrays.jl

### PR DESCRIPTION
**Pull request by tldrma**

### CODE REVIEW

File: test/array_of_similar_arrays.jl [hunk 1]
```diff
@@ -204,10 +203,12 @@ using UnsafeArrays
         A_nested = nestedview(A_flat, 2)
         @test A_nested isa AbstractArray{<:AbstractArray{T,2},3} where T
         @test flatview(A_nested) === A_flat
+#       @test VectorOfSimilarArrays(flatview(A_nested)) === A_nested
```
Comment (tldrma) [inline test/array_of_similar_arrays.jl]: This can be ignored. It was intended to be removed before push.

**Comment by tldrma** [comment]
Remaining tests to write for this PR:

- [ ] append! and prepend!

- [ ] similar

- [ ] setindex!

**Comment by oschulz** [comment]
Could you please put the StatsBase-specific tests (but only those) in a separate file, "test_statsbase_support.jl"? Basically all tests that test code in "statsbase_support.jl".

File: test/array_of_similar_arrays.jl [hunk 2]
```diff
@@ -198,7 +236,61 @@ using UnsafeArrays
...
+            @test @inferred(isless(diff, eps(Float64)))
+        end
 
+        a1 = a1 .- mu_a1; a2 = a2 .- mu_a2; a3 = a3 .- mu_a3
+        @testset "centered" begin
+            @test isapprox(mean(a1), 0, atol=eps(Float64))
+            @test isapprox(mean(a2), 0, atol=eps(Float64))
+            @test isapprox(mean(a3), 0, atol=eps(Float64))
+        end
+    end
```
Comment (tldrma) [inline test/array_of_similar_arrays.jl]: Does this include the calls to Statistics.cov, etc.? The ones that aren't using e.g. StatsBase.Weights and are thus not testing statsbase_support.jl? Specfically this part of the tests.

Comment (oschulz) [reply test/array_of_similar_arrays.jl]: Yes, the tests without weights should be in here.

File: test/array_of_similar_arrays.jl [hunk 3]
```diff
@@ -217,5 +279,36 @@ using UnsafeArrays
...
+        ASA = ArrayOfSimilarArrays([r1,r2,r3,r4])
+
+        f = x -> x.*2
+         
+        @test @inferred(IndexStyle(VSV)) == IndexLinear()
+        @test VSA == VSV
+        @test flatview(VSA) == flatview(VSV)
+
+        @test @inferred(deepmap(f, ASA)).data == ASA.data.*2
+        @test @inferred(innermap(f, ASA)).data == ASA.data.*2
```
[the remark this answers is not in the document]
Comment (oschulz) [reply test/array_of_similar_arrays.jl]: Yes, it would be good to test deepmap with a depth >2.

**Comment by oschulz** [comment]
Thanks!

**Pull request merged**